src/features/creators/newPackageProject.ts

```diff
@@ -1,18 +1,115 @@
 import { PythonProject, PythonProjectCreator, PythonProjectCreatorOptions } from '../../api';
-// import { runInBackground } from '../execution/runInBackground';
+import {
+    promptForVenv,
+    promptForCopilotInstructions,
+    isCopilotInstalled,
+    quickCreateNewVenv,
+    removeCopilotInstructions,
+    replaceInFilesAndNames,
+    replaceInFile,
+} from './creationHelpers';
+import { EXTENSION_ROOT_DIR } from '../../common/constants';
+import { EnvironmentManagers } from '../../internal.api';
 
 export class NewPackageProject implements PythonProjectCreator {
     public readonly name = 'newPackage';
     public readonly displayName = 'Package';
     public readonly description = 'Create a new Python package';
     public readonly tooltip = new MarkdownString('Create a new Python package');
 
-    constructor() {}
+    constructor(private readonly envManagers: EnvironmentManagers) {}
 
     async create(_options?: PythonProjectCreatorOptions): Promise<PythonProject | undefined> {
-        // show notification that the package creation was selected than return undefined
-        window.showInformationMessage('Creating a new Python package...');
-        return undefined;
+        // Prompt for package name
+        const packageName = await window.showInputBox({
+            prompt: 'What is the name of the package? (e.g. my_package)',
+            ignoreFocusOut: true,
+        });
+        if (!packageName) {
+            return undefined;
+        }
+
+        // Use helper to prompt for virtual environment creation
+        const createVenv = await promptForVenv();
+        if (createVenv === undefined) {
+            return undefined;
+        }
+
+        // Only prompt for Copilot instructions if Copilot is installed
+        let createCopilotInstructions = false;
+        if (isCopilotInstalled()) {
+            const copilotResult = await promptForCopilotInstructions();
+            if (copilotResult === undefined) {
+                return undefined;
+            }
+            createCopilotInstructions = copilotResult === true;
+        }
+
+        window.showInformationMessage(
+            `Creating a new Python project: ${packageName}\nvenv: ${createVenv}\nCopilot instructions: ${createCopilotInstructions}`,
+        );
+
+        // 1. Copy template folder
+        const templateFolder = path.join(
+            EXTENSION_ROOT_DIR,
+            'src',
+            'features',
+            'creators',
+            'templates',
+            'newPackageTemplate',
+        );
+        if (!(await fs.pathExists(templateFolder))) {
+            window.showErrorMessage('Template folder does not exist.');
+            return undefined;
+            // might need another check or error handling here
+        }
+        const workspaceFolders = workspace.workspaceFolders;
+        if (!workspaceFolders || workspaceFolders.length === 0) {
+            window.showErrorMessage('No workspace folder is open.');
```

Comment:
Are we able to open a workspace folder for them if it is empty using the logic Bhavya does with Create New?

Reply:
Bhayva should supply a path where the new project should be created- is the concern if that path doesn't exist?